components/esmterrain/storage.cpp

```diff
@@ -155,7 +155,7 @@ namespace ESMTerrain
                                             osg::ref_ptr<osg::Vec4Array> colours)
     {
         // LOD level n means every 2^n-th vertex is kept
-        size_t increment = 1 << lodLevel;
+        size_t increment = static_cast<int64_t>(1) << lodLevel;
```

Comment:
int64_t is not the same as size_t. Think about 32 bit systems.

Wouldn't this work?
<pre>
static_cast<size_t>(1) << lodLevel;
</pre>